homeassistant/components/pure_energie/__init__.py

```diff
@@ -0,0 +1,79 @@
+"""The Pure Energie integration."""
+from __future__ import annotations
+
+from typing import TypedDict
+
+from gridnet import Device, GridNet, SmartBridge
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DOMAIN, LOGGER, SCAN_INTERVAL, SERVICE_DEVICE, SERVICE_SMARTBRIDGE
+
+PLATFORMS = [Platform.SENSOR]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Pure Energie from a config entry."""
+
+    coordinator = PureEnergieDataUpdateCoordinator(hass)
+    try:
+        await coordinator.async_config_entry_first_refresh()
+    except ConfigEntryNotReady:
+        await coordinator.gridnet.close()
+        raise
+
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.entry_id] = coordinator
+
+    hass.config_entries.async_setup_platforms(entry, PLATFORMS)
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload Pure Energie config entry."""
+    unload_ok = await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+    if unload_ok:
+        del hass.data[DOMAIN][entry.entry_id]
+    return unload_ok
+
+
+class PureEnergieData(TypedDict):
```

Comment:
```suggestion
class PureEnergieData(NamedTuple):
```

If you make this a named tuple, it would be a little friendlier/easier to use in the rest of the code. E.g., `data.device` vs `data["device"]`